buyer-api/src/facades/helpers/performTransaction.js

```diff
@@ -81,12 +81,64 @@ const transactionResponse = async (web3, receipt) => {
 const sendSignedTransaction = (web3, signedTransaction) =>
   new Promise((resolve, reject) => {
     web3.eth.sendSignedTransaction(`0x${signedTransaction}`)
+      .on('error', (error, receipt) => {
+        logger.debug(`[perform tx] tx errored ${receipt}`);
+        reject(error);
+      })
+      .on('receipt', () => {
+        logger.debug('[perform tx] tx receipt done');
+      })
       .on('transactionHash', (hash) => {
         if (!hash) reject(new Error('No tx hash'));
+        logger.debug('[perform tx] tx hash done');
         resolve(hash);
       });
   });
 
+/**
+ * @param {Object} web3 instance of Web3
+ * @param {Sting} receipt transaction receipt
+ * @returns {Promise} promise that resolves to the transaction object in any of
+ *                    the following states: `pending`, `success` or `failure`.
+ *                    Is reject if an error occurs.
+ */
+const getTransaction = (web3, receipt) =>
+  new Promise((resolve, reject) => {
+    web3.eth.getTransactionReceipt(receipt, (err, result) => {
+      if (err) {
+        reject(err);
+      } else if (!result) {
+        resolve({ status: 'pending' });
+      } else if (result.status && result.status === 0x1) {
+        resolve({ ...result, status: 'success' });
+      } else {
+        resolve({ ...result, status: 'failure' });
+      }
+    });
+  });
+
+/**
+ * @param {Object} web3 instance of Web3
+ * @param {Sting} receipt transaction receipt
+ * @param {Object} opts
+ * @throws {Error} when `getTransaction` is rejected
+ */
+const waitForExecution = async (web3, receipt, opts = {}) => {
+  const { maxIterations = 20, Interval = 30 } = opts;
```

Comment:
Change casing of `Interval` to `interval`:
```suggestion
  const { maxIterations = 20, interval = 30 } = opts;
```